jung-visualization/src/test/java/edu/uci/ics/jung/visualization/spatial/rtree/RTreeTest.java

```diff
@@ -227,11 +227,23 @@ private void testAreas(TreeNode rootNode) {
     Rectangle2D rootBounds = rootNode.getBounds();
     if (rootNode instanceof InnerNode) {
       InnerNode innerNode = (InnerNode) rootNode;
-      Assert.assertEquals(rootBounds, Node.union(innerNode.getChildren()));
+      Assert.assertTrue(closeEnough(rootBounds, Node.union(innerNode.getChildren())));
     }
 
     for (TreeNode rt : rootNode.getChildren()) {
       testAreas(rt);
     }
   }
+
+  private boolean closeEnough(Rectangle2D left, Rectangle2D right) {
+    return left.equals(right)
+        || (closeEnough(left.getMinX(), right.getMinX())
```

Comment:
I would be inclined to use the Truth library for this; jbduncan@ and I will probably convert most/all of the JUNG test assertions to use Truth once things settle down a bit.  Specifically, Truth provides facilities for comparing double values within a specified tolerance: https://google.github.io/truth/floating_point

Reply:
sounds like something for a future PR. I like that idea.

Reply:
I like this idea too. I'll go and open an issue on this for now, if we don't already have one already, just so we don't lose track of it. :)

Reply:
@jbduncan I pushed a commit to use 'truth' in my latest PR. Please have a look to see if it can be improved. I noticed that there are a number of other places in jung-algorithms legacy code where a tolerance is used in a unit test comparison. Using 'truth' really cleans up the code. Kudos to the author(s) for making it so readable.

Reply:
@tomnelson I've had a look and made some suggestions! Please let me know what you think over there. :)